examples/grafana/grafana-ocp.yaml

```diff
@@ -26,51 +53,609 @@ objects:
+- apiVersion: extensions/v1beta1
+  kind: Deployment
+  metadata:
     labels:
-      metrics-infra: grafana-ocp
-      name: grafana-ocp
+      app: grafana-ocp
+    name: grafana-ocp
+    namespace: "${NAMESPACE}"
   spec:
-    selector:
-      name: grafana-ocp
     replicas: 1
+    selector:
+      matchLabels:
+        app: grafana-ocp
     template:
-      version: v1
       metadata:
         labels:
-          metrics-infra: grafana-ocp
-          name: grafana-ocp
+          app: grafana-ocp
+        name: grafana-ocp-app
       spec:
-        volumes:
-        - name: data
-          emptyDir: {}
+        serviceAccountName: grafana-ocp
         containers:
-        - image: "mrsiano/grafana-ocp:latest"
-          name: grafana-ocp
+        - name: oauth-proxy
+          image: ${IMAGE_PROXY}
+          imagePullPolicy: IfNotPresent
+          ports:
+          - containerPort: 8443
+            name: web
+          args:
+          - -https-address=:8443
+          - -http-address=
+          - -email-domain=*
+          - -client-id=system:serviceaccount:${NAMESPACE}:grafana-ocp
+          - -upstream=http://localhost:3000
+          - -provider=openshift
+#          - '-openshift-delegate-urls={"/api/datasources": {"resource": "namespace", "verb": "get", "resourceName": "grafana-ocp", "namespace": "${NAMESPACE}"}}'
+          - '-openshift-sar={"namespace": "${NAMESPACE}", "verb": "list", "resource": "services"}'
+          - -tls-cert=/etc/tls/private/tls.crt
+          - -tls-key=/etc/tls/private/tls.key
+          - -client-secret-file=/var/run/secrets/kubernetes.io/serviceaccount/token
+          - -cookie-secret-file=/etc/proxy/secrets/session_secret
+          - -skip-auth-regex=^/metrics,/api/datasources,/api/dashboards
+          volumeMounts:
+          - mountPath: /etc/tls/private
+            name: gf-tls
+          - mountPath: /etc/proxy/secrets
+            name: secrets
+
+        - name: grafana-ocp
+          image: ${IMAGE_GF}
           ports:
           - name: grafana-http
             containerPort: 3000
           volumeMounts:
-          - name: data
-            mountPath: "/root/go/src/github.com/grafana/grafana/data"
+          - mountPath: "/root/go/src/github.com/grafana/grafana/data"
+            name: gf-data
+          - mountPath: "/root/go/src/github.com/grafana/grafana/conf"
+            name: gfconfig
+          - mountPath: /etc/tls/private
+            name: gf-tls
+          - mountPath: /etc/proxy/secrets
+            name: secrets
           command:
-          - "./bin/grafana-server"
+           - "./bin/grafana-server"
+
+        volumes:
+        - name: gfconfig
+          configMap:
+            name: gf-config
+        - name: secrets
+          secret:
+            secretName: gf-proxy
+        - name: gf-tls
+          secret:
+            secretName: gf-tls
+        - emptyDir: {}
+          name: gf-data
+- apiVersion: v1
+  kind: ConfigMap
```

Comment:
@smarterclayton I am in favor of placing config in its own file and not as a ConfigMap. This example already has a setup script so we can create the configmap there. Thoughts?

Reply:
can you explain the advantages of it?

I see other templates also uses configmap i.e prometheus.yaml

Reply:
Separate file is better because it's easier to layer patches on top vs keeping it inside a template with other resources. Also, when the config file is bare (iow not inside a ConfigMap), it is easier to test since it's compatible with more tools. Going from a file to a CM is simple, the other way around is not when you are not inside a pod and `-o jsonpath='{.data}'` returns a distorted format.

> I see other templates also uses configmap i.e prometheus.yaml

I think that's in place because of the convenience of processing and creating from a template. Appropriate for an example but we also need to use these in our CI cluster from a single source (this dir) and the monolithic template approach does not work for us (reasons above). Since you already have a setup script here, I would like to see us take this approach.

Reply:
Discussed in IRC, we can leave this as is for now and address spliting the Grafana config to a separate file in a follow-up.